contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -570,6 +627,33 @@ impl InterchainTokenService {
         Ok(())
     }
 
+    fn execute_link_token_message(
```

Comment:
send and receive should have been separate PRs ideally, since this is already ~1000 lines review